ds3-sdk-integration/src/test/java/com/spectralogic/ds3client/integration/Job_management.java

```diff
@@ -0,0 +1,342 @@
+                    if (canceledJob.getId().equals(jobId)){
+                        isACanceledJob = true;
+                    }
+                }
+                assertTrue((System.nanoTime() - startTime)/1000000000 < testTimeOutSeconds );
+            }
+
+        } finally {
+            deleteAllContents(client, bucketName);
+        }
+    }
+
+    @Test
+    public void cancelAllJobs() throws IOException, SignatureException, XmlProcessingException {
+        final String bucketName = "test_cancel_all_jobs";
+        try {
+            client.putBucket(new PutBucketRequest(bucketName));
+
+            final List<Ds3Object> objectsOne = new ArrayList<>();
+            final Ds3Object obj = new Ds3Object("testOne", 2);
+            objectsOne.add(obj);
+
+            final Ds3ClientHelpers.Job jobOne =
+                    wrap(client).startWriteJob(bucketName, objectsOne);
+
+            final List<Ds3Object> objectsTwo = new ArrayList<>();
+            final Ds3Object objTwo = new Ds3Object("testTwo", 2);
+            objectsTwo.add(objTwo);
+
+            final Ds3ClientHelpers.Job jobTwo =
+                    wrap(client).startWriteJob(bucketName, objectsTwo);
+
+            final CancelAllJobsSpectraS3Response response = client
+                    .cancelAllJobsSpectraS3(new CancelAllJobsSpectraS3Request());
+            response.checkStatusCode(204);
+
+            assertTrue(client.getActiveJobsSpectraS3(new GetActiveJobsSpectraS3Request())
+                    .getActiveJobListResult().getActiveJobs().isEmpty());
+            } finally {
+            deleteAllContents(client, bucketName);
+        }
+    }
+
+    //todo public void cancelAllJobs() without force and 1 object uploaded
+    //todo public void cancelAllJobs() with force and 1 object uploaded
+
+    @Test
+        public void getCanceledJobs() throws IOException, SignatureException, XmlProcessingException {
+            final String bucketName = "test_get_canceled_jobs";
+            try {
+                client.putBucket(new PutBucketRequest(bucketName));
+
+                final List<Ds3Object> objectsOne = new ArrayList<>();
+                final Ds3Object obj = new Ds3Object("testOne", 2);
+                objectsOne.add(obj);
+
+                final Ds3ClientHelpers.Job jobOne =
+                        wrap(client).startWriteJob(bucketName, objectsOne);
+                final UUID jobOneId = jobOne.getJobId();
+                final CancelJobSpectraS3Response cancelResponse = client
+                        .cancelJobSpectraS3(new CancelJobSpectraS3Request(jobOneId));
+
+                final GetCanceledJobsSpectraS3Response getCanceledJobsResponse = client
+                        .getCanceledJobsSpectraS3(new GetCanceledJobsSpectraS3Request());
+
+                for (CanceledJob job : getCanceledJobsResponse.getCanceledJobListResult().getCanceledJobs()) {
```

Comment:
This method for checking for the desired ID is not very clear in meaning. It would be more clear to create a list of all IDs, and then write an assert to check if the list of IDs contains the desired ID